wikimedia/wikipedia-ios
# Add Notifications Center onboarding modal and panel

**Phabricator:** https://phabricator.wikimedia.org/T302633 (partially)

### Notes
This PR adds the onboarding UI features to the Notifications Center (specifically, the first launch "Editing notifications" modal and the push opt in panel in the notifications center itself). 

Fully completing this task is blocked by:
- [FAQ on Meta to link to](https://phabricator.wikimedia.org/T302634) 
- Actually opting the user into system/echo notifications if they select that option in the panel. I recommend following on to this PR once the work from https://github.com/wikimedia/wikipedia-ios/pull/4138 lands.

I don't think those blockers should prevent code review from happening on this, but up to you @tonisevener on if you want to merge this or not after review until it's completely feature complete in follow-up PRs.

### Test Steps
1. Be sure you are logged in
2. Tap the Notifications Center bell. Confirm you see the onboarding full screen modal.
3. Confirm tapping the "Learn more" button sends you to external Safari.
4. Tap Continue in the modal and confirm you see a Turn on push notifications opt in panel.
5. Select any option or tap to dismiss, then dismiss the Notifications Center.
6. Tap the Notifications Center bell again and confirm you do not see the onboarding modal or panel.

To speed up testing cycles, it may be helpful to add this to the `viewDidLoad` of `NotificationsCenterViewController.swift` to always display the modal and panel on Notifications Center launch:
```swift
UserDefaults.standard.removeObject(forKey: "WMFUserHasOnboardedToNotificationsCenter")
```
